test/functional/mempool_limit.py

```diff
@@ -20,55 +23,76 @@ def set_test_params(self):
-
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
-        base_fee = relayfee*100
-        for i in range (3):
-            txids.append([])
-            txids[i] = create_lots_of_big_transactions(self.nodes[0], txouts, utxos[30*i:30*i+30], 30, (i+1)*base_fee)
+        tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node = node, fee_rate = relayfee)['txid']
+
+        base_fee = relayfee*1000
+        # We increase the tx fee massively now to give these transactions a higher priority in the mempool
+
+        for batch_of_txid in range (3):
+            tx_ids.append([])
+            tx_ids[batch_of_txid] = self.create_large_transactions(node, txouts, miniwallet, 30, (batch_of_txid+1)*base_fee)
+            # We gradually increase the tx fee by incrementing it by a factor of (basee_fee) for each batch of 30 transactions
 
         self.log.info('The tx should be evicted by now')
-        assert txid not in self.nodes[0].getrawmempool()
-        txdata = self.nodes[0].gettransaction(txid)
-        assert txdata['confirmations'] ==  0  #confirmation should still be 0
+        assert_greater_than(len([txid for batch in tx_ids for txid in batch]), len(node.getrawmempool()))
+        # We assert that the number of created transactions is greater than the ones in the mempool
+        assert tx_to_be_evicted_id not in node.getrawmempool() 
+        # We assert that our initial tx_to_be_evicted_id is not present in the mempool anymore as it's fee was less
 
         self.log.info('Check that mempoolminfee is larger than minrelytxfee')
-        assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
-        assert_greater_than(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_greater_than(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
         self.log.info('Create a mempool tx that will not pass mempoolminfee')
-        us0 = utxos.pop()
-        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
-        outputs = {self.nodes[0].getnewaddress() : 0.0001}
-        tx = self.nodes[0].createrawtransaction(inputs, outputs)
-        # specifically fund this tx with a fee < mempoolminfee, >= than minrelaytxfee
-        txF = self.nodes[0].fundrawtransaction(tx, {'feeRate': relayfee})
-        txFS = self.nodes[0].signrawtransactionwithwallet(txF['hex'])
-        assert_raises_rpc_error(-26, "mempool min fee not met", self.nodes[0].sendrawtransaction, txFS['hex'])
+        # We deliberately create a tx with a fee less that the minimum mempool fee to assert that it does not get added to the mempool
+        assert_raises_rpc_error(-26, "mempool min fee not met", miniwallet.send_self_transfer, from_node=node, fee_rate=relayfee,mempool_valid=False)
+
+    def create_large_transactions(self, node, array_of_large_tx, miniwallet, no_of_tx_ids, fee_rate):
```

Comment:
conventionally the helper function goes before the main test function. poke around some other test files to see examples